static/app/components/modals/inviteMembersModal/inviteRowControlNew.spec.tsx

```diff
@@ -25,74 +25,108 @@ describe('InviteRowControlNew', function () {
   ];
   const teams = teamData.map(data => TeamFixture(data));
 
-  const rowControlProps: ComponentProps<typeof InviteRowControlNew> = {
-    disabled: false,
-    emails: [],
-    inviteStatus: {},
-    onChangeEmails: mockOnChangeEmails,
-    onChangeRole: mockOnChangeRole,
-    onChangeTeams: mockOnChangeTeams,
-    onRemove: () => {},
-    role: '',
-    roleDisabledUnallowed: false,
-    roleOptions: [
-      {
-        id: 'member',
-        name: 'Member',
-        desc: '...',
-        minimumTeamRole: 'contributor',
-        isTeamRolesAllowed: true,
-      },
-      {
-        id: 'billing',
-        name: 'Billing',
-        desc: '...',
-        minimumTeamRole: 'contributor',
-        isTeamRolesAllowed: false,
-      },
-    ],
-    teams: [],
-  };
+  const getComponent = (role?: string) => (
+    <InviteMembersContext.Provider
+      value={{
+        complete: false,
+        inviteStatus: {},
+        invites: [],
+        pendingInvites: [
+          {
+            emails: new Set<string>(),
+            teams: new Set<string>(),
+            role: role || '',
+          },
+        ],
+        reset: () => {},
+        sendInvites: () => {},
+        sendingInvites: false,
+        setEmails: mockOnChangeEmails,
+        setRole: mockOnChangeRole,
+        setTeams: mockOnChangeTeams,
+        willInvite: true,
```

Comment:
can simplify these tests to only what you want to test and allow the context defaults to set everything else.